src/test/java/com/scalar/database/storage/cassandra4driver/ClusterManagerTest.java

```diff
@@ -0,0 +1,224 @@
+        .doesNotThrowAnyException();
+
+    // Assert
+    verify(manager).getBuilder();
+    verify(builder).build();
+  }
+
+  @Test
+  public void getSession_ConfigWithInvalidPortGiven_ShouldThrowExceptionProperly() {
+    // Arrange
+    props.setProperty(DatabaseConfig.CONTACT_PORT, INVALID_PORT);
+    config = new DatabaseConfig(props);
+    manager = Mockito.spy(new ClusterManager(config));
+
+    // Act Assert
+    assertThatThrownBy(
+            () -> {
+              manager.getSession();
+            })
+        .isInstanceOf(ConnectionException.class)
+        .hasCauseExactlyInstanceOf(IllegalArgumentException.class);
+
+    // Assert
+    verify(builder, never()).build();
+  }
+
+  @Test
+  public void getSession_CalledOnceAndDriverExceptionThrown_ShouldThrowConnectionException() {
+    // Arrange
+    DriverException toThrow = mock(DriverException.class);
+    when(builder.build()).thenThrow(toThrow);
+
+    // Act Assert
+    assertThatThrownBy(
+            () -> {
+              manager.getSession();
+            })
+        .isInstanceOf(ConnectionException.class)
+        .hasCause(toThrow);
+
+    // Assert
+    verify(manager).getBuilder();
+    verify(builder).build();
+  }
+
+  @Test
+  public void getSession_CalledAfterBuilt_ShouldNotBuild() {
+    // Arrange
+    manager.getSession();
+    manager.close();
+
+    // Act Assert
+    assertThatCode(
+            () -> {
+              manager.getSession();
+            })
+        .doesNotThrowAnyException();
+
+    // Assert
+    verify(manager, times(1)).getBuilder();
+    verify(builder, times(2)).build();
+  }
+
+  @Test
+  public void getSession_CalledTwice_ShouldReuseSession() {
+    // Arrange
+    manager.getSession();
+
+    // Act Assert
+    assertThatCode(
+            () -> {
+              manager.getSession();
+            })
+        .doesNotThrowAnyException();
+
+    // Assert
+    verify(manager, times(1)).getBuilder();
+    verify(builder, times(1)).build();
+  }
+
+  @Test
+  public void constructor_NullGiven_ShouldThrowNullPointerException() {
+    // Act Assert
+    assertThatThrownBy(
+            () -> {
+              new ClusterManager(null);
+            })
+        .isInstanceOf(NullPointerException.class);
+  }
+
+  @Test
+  public void getMetadata_ExistingKeyspaceAndTableGiven_ShouldReturnMetadata() {
+    // Arrange
+    manager.getSession();
+    Metadata metadata = mock(Metadata.class);
+    KeyspaceMetadata keyspaceMetadata = mock(KeyspaceMetadata.class);
+    TableMetadata tableMetadata = mock(TableMetadata.class);
+    when(session.getMetadata()).thenReturn(metadata);
+    when(metadata.getKeyspace(anyString())).thenReturn(Optional.of(keyspaceMetadata));
+    when(keyspaceMetadata.getTable(anyString())).thenReturn(Optional.of(tableMetadata));
+
+    // Act
+    TableMetadata actual = manager.getMetadata(ANY_KEYSPACE_NAME, ANY_TABLE_NAME);
+
+    // Assert
+    assertThat(actual).isEqualTo(tableMetadata);
+  }
+
+  // @Test
```

Comment:
What is the reason of keeping this commented out  ?
If we want to keep in with this form, it would probably be nice to add some comments about why it is here.

Reply:
`getMetadata()` doesn't throw an exception for now.
https://docs.datastax.com/en/drivers/java/4.3/com/datastax/oss/driver/api/core/session/Session.html#getMetadata--
Removed the test.